specification/azsadmin/resource-manager/storage/Microsoft.Storage.Admin/preview/2019-08-08/storageaccounts.json

```diff
@@ -0,0 +1,593 @@
+            "$ref": "storageaccounts.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "storageaccounts.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Garbage collection has completed."
+          },
+          "202": {
+            "description": "Accepted -- on demand gc operation triggered; operation will complete asynchronously."
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    }
+  },
+  "definitions": {
+    "StorageAccountState": {
+      "readOnly": true,
+      "description": "Storage account state.",
+      "type": "string",
+      "enum": [
+        "Creating",
+        "Succeeded",
+        "Suspended"
+      ],
+      "x-ms-enum": {
+        "name": "StorageAccountState",
+        "modelAsString": true
+      }
+    },
+    "StorageAccountType": {
+      "description": "Storage account type.",
+      "type": "string",
+      "enum": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_ZRS",
+        "Standard_RAGRS",
+        "Premium_LRS"
+      ],
+      "x-ms-enum": {
+        "name": "StorageAccountType",
+        "modelAsString": true
+      }
+    },
+    "LocationStatus": {
+      "readOnly": true,
+      "type": "string",
+      "description": "Gets the status indicating whether the primary location of the storage account is available or unavailable.",
+      "enum": [
+        "Available",
+        "Unavailable"
+      ],
+      "x-ms-enum": {
+        "name": "LocationStatus",
+        "modelAsString": true
+      }
+    },
+    "StorageAccountStatus": {
+      "description": "The state of storage account in WAC.",
+      "type": "string",
+      "enum": [
+        "Active",
+        "Deleted"
+      ],
+      "x-ms-enum": {
+        "name": "StorageAccountStatus",
+        "modelAsString": true
+      }
+    },
+    "WacAccountPermissions": {
+      "description": "Current operation being performed on Storage Account",
+      "type": "string",
+      "enum": [
+        "Empty",
+        "None",
+        "Read",
+        "Delete",
+        "Write",
+        "Full"
+      ],
+      "x-ms-enum": {
+        "name": "WacAccountPermissions",
+        "modelAsString": true
+      }
+    },
+    "WacAccountStates": {
+      "description": "Current operation being performed on Storage Account",
+      "type": "string",
+      "enum": [
+        "None",
+        "Active",
+        "Deleted",
+        "AboveQuota",
+        "Suspended",
+        "All",
+        "GcReadyState"
+      ],
+      "x-ms-enum": {
+        "name": "WacAccountStates",
+        "modelAsString": true
+      }
+    },
+    "StorageEncryptionSetting": {
+      "description": "Storage Encryption setting",
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "Bool to indicate whether enabled or not.",
+          "type": "boolean",
```

Comment:
We try to not use boolean as a type. They can only ever have 2 values so it requires a new api version to extend it.

String based enums are preferred even if they contain 'true' and 'false'. In your case it might be better to have something like "state": "enabled"/"disable" instead of true/false.

Reply:
The type for enabled will never be changed, so we consider boolean enough here.